yais/src/Memory.cc

```diff
@@ -142,4 +143,24 @@ std::string BytesToString(size_t bytes)
     return std::string(buffer);
 }
 
-} // namespace yais
\ No newline at end of file
+std::uint64_t StringToBytes(const std::string str)
+{
+    // https://regex101.com/r/UVm5wT/1
+    std::smatch m;
+    std::regex r("(\\d+[.\\d+]*)([KMGTkmgt]*)([i]*)[bB]");
+    std::map<char, int> prefix = {
+        {'k', 1}, {'m', 2}, {'g', 3}, {'t', 4},
+        {'K', 1}, {'M', 2}, {'G', 3}, {'T', 4},
+    };
+
+    if (!std::regex_search(str, m, r))
+	LOG(FATAL) << "Unable to convert \"" << str << "\" to bytes. "
+                   << "Expected format: 10b, 1024B, 1KiB, 10MB, 2.4gb, etc.";
+
+    std::uint64_t base = m[3] == "" ? 1000 : 1024;
```

Comment:
If `m` is empty, this would try to access a unknown place in memory. Should the check be:
```cpp
const std::uint64_t base = m.empty() || (m.size() > 3 && m[3] == "") ? 1000 : 1024;
```